client/policy_validator.go

```diff
@@ -348,9 +350,22 @@ func (p *PolicyAPIClient) execGet(args ...string) (*PolicyAPIResult, error) {
 	fullArgs := []string{"get", "policies"}
 	fullArgs = append(fullArgs, args...)
 	fullArgs = append(fullArgs, "-o", "json")
-	out, err := p.cli.exec(fullArgs, p.cli.GetNamespace())
+	var out *bytes.Buffer
+	err = utils.RetryWithContext(ctx, time.Millisecond*100, func() (bool, error) {
+		out, err = p.cli.exec(fullArgs, p.cli.GetNamespace())
+		if err != nil {
+			if _, err := getRootObject(p.cli); err != nil && errors.IsNotFound(err) {
+				return true, notEnabledErr
+			}
+			return false, nil
+		}
+		return true, nil
+	})
 	if err != nil {
-		return nil, fmt.Errorf("Policy validation error: %v", err)
+		if os.IsTimeout(err) {
+			return nil, fmt.Errorf("Policy validation error: %s not ready", types.ControllerDeploymentName)
```

Comment:
Any chance capturing and showing the last error that ocurred, in addition to mentioning the timeout?  That might help in investigations

Reply:
I was thinking about it as well. I think that after 30s if it is still failing, it should be useful.
I will include it.